src/backend/db/database_manager.py

```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Retrieve comment details
+            comment_query = '''
+            SELECT comment_id, ticket_id, posted_by, comment, created_at, is_edited
+                FROM ticket_comments
+                WHERE comment_id = :comment_id AND ticket_id = :ticket_id
+            '''
+            comment_result = session.execute(
+                text(comment_query),
+                {"comment_id": comment_id, "ticket_id": ticket_id},
+            ).fetchone()
+
+            if not comment_result:
+                return None  # If comment is not found, return None
+
+            # Convert comment result into a dictionary
+            logger.info(f"Comment details by comment_id {comment_result}")
+            comment_data = {
+                column: value
+                for column, value in zip(comment_result.keys(), comment_result)
+            }
+
+            return comment_data
+
+        except (OperationalError, DatabaseError) as e:
+            logger.error(f"Database error: {e}")
+            raise Exception("Database connectivity issue")
+        except SQLAlchemyError as e:
+            logger.error(f"SQLAlchemy error: {e}")
+            raise Exception("Database query error")
+        finally:
+            session.close()
+
+    def get_paginated_comments_by_ticket_id(self, customer_guid, ticket_id, page=1, page_size=10):
+        logger.debug("Entering get_paginated_comments_by_ticket_id method")
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+
+        try:
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the ticket exists
+            ticket_exists = session.execute(
+                text("SELECT 1 FROM tickets WHERE ticket_id = :ticket_id LIMIT 1"),
+                {"ticket_id": ticket_id}
+            ).fetchone()
+
+            if not ticket_exists:
+                logger.error(f"Invalid ticket_id: {ticket_id}")
+                return []
+
+            # Pagination logic
+            offset = (page - 1) * page_size
+            logger.debug(f"Fetching comments with pagination: page={page}, page_size={page_size}, offset={offset}")
+
+            query = """
+                SELECT comment_id, comment, created_at
+                FROM ticket_comments
+                WHERE ticket_id = :ticket_id
+                ORDER BY created_at ASC
+                LIMIT :page_size OFFSET :offset
+            """
+            results = session.execute(
+                text(query),
+                {"ticket_id": ticket_id, "page_size": page_size, "offset": offset},
+            ).fetchall()
+
+            comments_list = [
+                {"comment_id": row.comment_id, "comment": row.comment, "created_at": row.created_at}
```

Comment:
The output should have
            CREATE TABLE IF NOT EXISTS ticket_comments (
                comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
                ticket_id BIGINT NOT NULL,
                posted_by VARCHAR(255) NOT NULL,
                comment TEXT NOT NULL,
                is_edited BOOLEAN DEFAULT FALSE,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
            );

Reply:
modified sir